src/drv_oculus_rift/rift.c

```diff
@@ -341,39 +889,65 @@ static ohmd_device* open_device(ohmd_driver* driver, ohmd_device_desc* desc)
 	dump_packet_sensor_config(&priv->sensor_config);
 
 	// Set default device properties
-	ohmd_set_default_device_properties(&priv->base.properties);
+	ohmd_set_default_device_properties(&hmd_dev->base.properties);
 
 	// Set device properties
-	priv->base.properties.hsize = priv->display_info.h_screen_size;
-	priv->base.properties.vsize = priv->display_info.v_screen_size;
-	priv->base.properties.hres = priv->display_info.h_resolution;
-	priv->base.properties.vres = priv->display_info.v_resolution;
-	priv->base.properties.lens_sep = priv->display_info.lens_separation;
-	priv->base.properties.lens_vpos = priv->display_info.v_center;
-	priv->base.properties.ratio = ((float)priv->display_info.h_resolution / (float)priv->display_info.v_resolution) / 2.0f;
+	hmd_dev->base.properties.hsize = priv->display_info.h_screen_size;
+	hmd_dev->base.properties.vsize = priv->display_info.v_screen_size;
+	hmd_dev->base.properties.hres = priv->display_info.h_resolution;
+	hmd_dev->base.properties.vres = priv->display_info.v_resolution;
+	hmd_dev->base.properties.lens_sep = priv->display_info.lens_separation;
+	hmd_dev->base.properties.lens_vpos = priv->display_info.v_center;
+	hmd_dev->base.properties.ratio = ((float)priv->display_info.h_resolution / (float)priv->display_info.v_resolution) / 2.0f;
+
+	if (desc->revision == REV_CV1) {
+		hmd_dev->base.properties.control_count = 9;
```

Comment:
This could use a comment that these are for the remote control. I was seriously confused for a moment.

Reply:
Done